src/test/java/org/opentripplanner/apis/gtfs/GraphQLIntegrationTest.java

```diff
@@ -212,6 +230,21 @@ public TransitAlertService getTransitAlertService() {
     };
     routes.forEach(transitService::addRoutes);
 
+    // Crate a calendar (needed for testing cancelled trips)
+    CalendarServiceData calendarServiceData = new CalendarServiceData();
+    var firstDate = LocalDate.of(2024, 8, 8);
+    var secondDate = LocalDate.of(2024, 8, 9);
+    calendarServiceData.putServiceDatesForServiceId(cal_id, List.of(firstDate, secondDate));
+    transitModel.getServiceCodes().put(cal_id, 0);
+    transitModel.updateCalendarServiceData(true, calendarServiceData, DataImportIssueStore.NOOP);
+    transitModel.initTimetableSnapshotProvider(() -> {
+      TimetableSnapshot timetableSnapshot = new TimetableSnapshot();
+      tripTimes2.cancelTrip();
+      timetableSnapshot.update(pattern, tripTimes2, secondDate);
+
+      return timetableSnapshot.commit();
+    });
```

Comment:
```suggestion
    
    tripTimes2.cancelTrip();
    var timetableSnapshot = new TimetableSnapshot();
    timetableSnapshot.update(pattern, tripTimes2, secondDate);
    var committedSnapshot = timetableSnapshot.commit();
    transitModel.initTimetableSnapshotProvider(() -> committedSnapshot);
```

Reply:
This looks like a race condition that is triggered only in Linux (the Windows build does not fail)